src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1780,6 +1780,14 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                     is_allocatable, is_const, raise_error, abi, is_argument);
                 return ASRUtils::TYPE(ASR::make_Pointer_t(al, loc, type));
             } else if (var_annotation == "Const") {
+                ASR::ttype_t *type = ast_expr_to_asr_type(loc, *s->m_slice,
+                    is_allocatable, is_const, raise_error, abi, is_argument);
+                if (ASR::is_a<ASR::Tuple_t>(*type)) {
+                    throw SemanticError("'Const' not required as tuples are already immutable", loc);
+                }
+                else if (ASR::is_a<ASR::Character_t>(*type)) {
+                    throw SemanticError("'Const' not required as strings are already immutable", loc);
+                }
```

Comment:
@kmr-srbh I understand that `tuple` and `strings` are by default immutable in python. I am just curious, do we have any issue which mentions that explicit `Const` for `tuple` and `character` type should not be allowed?

Reply:
@Shaikh-Ubaid This is related only to design. We annotate a variable with `const` to essentially make it immutable in other programming languages. As `tuple` and `str` are immutable by design, it is redundant to annotate them again with `Const`.

This had made for a big relief with the `Const` node which existed before as we no more had to handle `tuple` and `str` methods on a `Const` type. That should probably not be the case now, but the redundancy remains.

Reply:
It looks like `Const` node for `tuple` and `str` can still exist. Consider the following example:

```console
from lpython import i32

a: Const[str] = "abc"
a = "pqr"
```

The above example works with `cpython`. We expect it to not work with `lpython`. `lpython` should ideally throw an error that an attempt to modify a `const` variable is being made.

I think `str` being immutable means updating elements of the string is not allowed, but updating the variable itself is allowed. That is, `a[0] = "x"` is not allowed, but `a = "pqr"` is allowed. 

So to not allow `a = "pqr"`, I think we will need the `Const[str]` type.

Thus, I would say for now, let's not close any doors for us by explicitly not allowing `Const[str]` or `Const[tuple[]]`.

Reply:
Ah! I get what you mean. I am reverting this change. :+1: